applications/FluidDynamicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_periodic.h

```diff
@@ -134,32 +134,53 @@ class ResidualBasedBlockBuilderAndSolverPeriodic
         // Assign an Equation Id to all non-duplicate nodes
         unsigned int EqId = 0;
 
+        // Modify edge/corner periodic condtions so that one of the nodes gets EquationId
+        for (ModelPart::ConditionIterator itCond = r_model_part.ConditionsBegin(); itCond != r_model_part.ConditionsEnd(); itCond++) {
+            auto& r_geometry = itCond->GetGeometry();
+            if (itCond->Is(PERIODIC) && (r_geometry.PointsNumber() == 4 || r_geometry.PointsNumber() == 8)) {
+                r_geometry[0].GetSolutionStepValue(mPeriodicIdVar) = 0;
+            }
+        }
+
         for (typename DofsArrayType::iterator itDof = BaseType::mDofSet.begin(); itDof != BaseType::mDofSet.end(); ++itDof)
         {
             if ( itDof->GetSolutionStepValue(mPeriodicIdVar) < static_cast<int>(itDof->Id()) )
                 itDof->SetEquationId(EqId++);
         }
 
+        // Reset edge/corner periodic condtions
+        for (ModelPart::ConditionIterator itCond = r_model_part.ConditionsBegin(); itCond != r_model_part.ConditionsEnd(); itCond++) {
```

Comment:
```suggestion
        for (auto itCond = r_model_part.ConditionsBegin(); itCond != r_model_part.ConditionsEnd(); itCond++) {
```

Idem with the it_cond